### PR TITLE
KAFKA-8090: Use automatic RPC generation in ControlledShutdown

### PR DESCRIPTION
**Pull request by mimaison**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mimaison** [comment]
@omkreddy @cmccabe If you have some time, can you take a look? Thanks

File: core/src/main/scala/kafka/server/KafkaServer.scala [hunk 1]
```diff
@@ -516,19 +517,21 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
                 else if (config.interBrokerProtocolVersion < KAFKA_2_2_IV0) 1
                 else 2
 
-              val controlledShutdownRequest = new ControlledShutdownRequest.Builder(config.brokerId,
-                kafkaController.brokerEpoch, controlledShutdownApiVersion)
+              val controlledShutdownRequest = new ControlledShutdownRequest.Builder(
```
Comment (omkreddy) [inline core/src/main/scala/kafka/server/KafkaServer.scala:520]: we need to pass ```controlledShutdownApiVersion``` value to the Builder to retain previous behavior ?

Comment (mimaison) [reply core/src/main/scala/kafka/server/KafkaServer.scala:520]: Thanks for the review. Yes you are right! I've pushed an update

**Review by omkreddy — APPROVED** [review]
@mimaison Thanks for the PR. LGTM.

**Pull request closed without merging**